lib/checkPullRequestTemplate.js

```diff
@@ -0,0 +1,240 @@
+    -pull-requests/allowing-changes-to-a-pull-request-branch-created-
+    from-a-fork) for instructions on how to enable it.)  - This
+    lets reviewers restart your CircleCI tests for you.
+    - [ ] The PR is made from a branch that's **not** called
+    "develop".
+  */
+  const checklistHeadingIndex = pullRequest.body.indexOf(CHECKLIST_TEXT);
+
+  if (checklistHeadingIndex !== -1) {
+    const nextHeadingIndex = pullRequest.body.indexOf(
+      '## ',
+      checklistHeadingIndex + CHECKLIST_TEXT.length
+    );
+
+    let checklistSection = pullRequest.body.substring(
+      checklistHeadingIndex,
+      nextHeadingIndex
+    );
+
+    // Get all unchecked items: they should start with "-[ ]".
+    const uncheckedItems = checklistSection.split('\n').filter((list) => {
+      return list.startsWith(UNCHECKED_CHECKLIST);
+    });
+
+    let message = '';
+
+    uncheckedItems.forEach((item) => {
+      // Check if karma/linter check item is unchecked.
+      if (item.toLowerCase().includes(KARMA_LINTER)) {
+        // Karma and linter is unchecked.
+
+        message +=
+          'The karma and linter checklist has not been checked, ' +
+          'please make sure to run the frontend tests and lint tests before ' +
+          'pushing. ';
+      }
+      // Check if edits from maintainers check item is unchecked.
+      if (item.toLowerCase().includes(EDITS_FROM_MAINTAINERS_TEXT)) {
+        // Check if maintainer can actually modify the PR. This is because
+        // it is possible for the maintainers to be able to modify the PR
+        // but the PR author forgot to check it in the description.
+        if (!pullRequest.maintainer_can_modify) {
+          message +=
+            'The allow edits from maintainers checklist needs to ' +
+            'be ticked so that maintainers can rerun failed tests. Endeavour ' +
+            'to add this by ticking on the check box. ';
+        }
+      }
+    });
+
+    return message;
+  } else {
+    return (
+      'The body of this PR is missing the checklist section, please ' +
+      'update it to include the checklist.'
+    );
+  }
+};
+
+/**
+ * @param {import('probot').Context} context
+ */
+const checkTemplate = async (context) => {
+  /**
+   * @type {import('probot').Octokit.PullsGetResponse} pullRequest
+   */
+  const pullRequest = context.payload.pull_request;
+
+  const explanationMessage = checkPRExplanation(pullRequest);
+  const checklistMessage = checkEssentialChecklist(pullRequest);
+  let commentBody = '';
+  if (explanationMessage && checklistMessage) {
+    commentBody =
+      'Hi @' +
+      pullRequest.user.login +
+      ', ' +
+      explanationMessage +
+      '<br>Also, ' +
+      swapCharInPlace(
+        checklistMessage,
+        0,
+        checklistMessage[0].toLowerCase()
+      ).trim() +
+      ' Thanks!';
+  } else if (explanationMessage) {
+    commentBody =
+      'Hi @' +
+      pullRequest.user.login +
+      ', ' +
+      swapCharInPlace(
```

Comment:
Why are we doing this - can we return the message directly with first char in lowercase?